### PR TITLE
chore: update testing action node version to align with package.json

### PR DESCRIPTION
**Pull request by jusx**
## Changes
- chore: update testing workflow node version to align with package.json
- fix: the npm command test-coverage invokes an upload to codecov. Remove redundant GH action to upload which is a dup.
- fix: to invoke CI on push in a PR branch

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -27,11 +24,3 @@ jobs:
     - run: npm run test-coverage
       env:
         CI: true
-    - name: Upload coverage to Codecov
```
Comment (shine2lay) [inline .github/workflows/testing.yml:30]: i don't see the duplication part of this code (mentioned in the description). are we simply removing this?

**Review by shine2lay — COMMENTED** [review]
left a comment

File: .github/workflows/testing.yml (hunk 1 above)
Comment (jusx) [reply .github/workflows/testing.yml:30]: The npm command includes an upload by calling codecov. This Github action does the same thing. In that respect, it's a duplication of task.

**Review by shine2lay — APPROVED** [review]

File: .github/workflows/testing.yml (hunk 1 above)
Comment (shine2lay) [reply .github/workflows/testing.yml:30]: ah okay, got it

**Pull request merged**